src/olmo_core/train/callbacks/evaluator_callback.py

```diff
@@ -172,6 +179,9 @@ def _perform_eval(self):
         self.trainer.record_metric("throughput/in-loop eval time (s)", total_time)
         self.trainer.record_metric("throughput/in-loop eval batches", total_bs)
 
+        if self.cancel_after_first_eval:
+            self.trainer.cancel_run("canceled from evaluator callback", no_sync=True)
```

Comment:
Maybe a clearer message, like "canceled because `cancel_after_first_eval` is set"?

Reply:
Done